### DISCUSSION TITLE
How to apply infinite scroll in budibase component?

### DISCUSSION THREAD

**Discussion by helicop0420** [Budibase/budibase#10522] [Help] [answered]
I use data provider and repeater to show the list.
but I want to apply infinite scroll to the list.
Could you tell me about the way to integrate infinite scroll?

**Comment by ConorWebb96** [comment]
Hey @wow0420,

I'm pretty sure to do this you would need to create a custom component. I've found a discord thread asking a similar question to the one you have. https://discord.com/channels/733030666647765003/1092540636739162112/1092540636739162112

Something of note, the new grid view supports scrolling pagination. Therefore I would assume that this will eventually get added to the design area as well.

Currently I don't believe the components within Budibase support infinite scrolling. I would suggest looking for or making a custom component. Then use the already existing bookmark pagination functionality to achieve the desired effect. For help with setting up your environment to build custom components look here: https://docs.budibase.com/docs/custom-plugin

I hope this makes sense.

**Reply by helicop0420** [reply]
Thanks. @ConorWebb96 ,
I found the channel, but there is no detailed way to develop custom component in budibase.
Maybe could you tell me about the way?

**Reply by ConorWebb96** [reply, marked as answer]
The documentation is mainly around setting up to build custom components. The rest is left to the person building these plugins. Here is a good video to get you started https://vimeo.com/754759041?embedded=true&source=video_title&owner=127542078.

This shows the basics, how to setup and discussed useful aspects e.g. variables, schema.json and bindings.

I'd also recommend looking at: https://github.com/Budibase/plugins as theres lots of different plugins which have been made already and its quite good to have a look around to aid in understanding.

**Reply by helicop0420** [reply]
@ConorWebb96  Thanks for your help!
one question.
I would like to make custom component named 'custom repeater' to apply infinite scroll.
when I am going to integrate infinite scroll to 'custom repeater',
the 'custom repeater' has to be independent from data provider or query, and display data from state variable.
Also once data provider receives new data from query, update state variable with new data (maybe array).
That's my idea to solve this problem, but not make sure if this is possible on budibase.
Hope your opinions...

**Reply by ConorWebb96** [reply]
I was thinking you could provide it with a table option in the schema.json, select the table you wish to use. 

Then with the selected table you could use the Budibase SDK API and query results with pagination set to true. From here, each time you scroll to the bottom of the container you can hook into the bookmark functionality to then query the next x number of rows. This could be done infinitely until you get to the end.

**Reply by helicop0420** [reply]
@ConorWebb96  thanks.
![image](https://github.com/Budibase/budibase/assets/116023731/b3816a1c-b611-47d7-b8db-756c8601624c)
Can you see above link?
I made table option on my custom component.
but how to use Budibase SDK API? and how to set pagination?
Also you mean Custom Component has to receive data from this table and table data has to be updated when scrolling?
I can't make sure how to update table- original data + new data from API for infinite-scrolling.
Could you explain in detail? or Can we have a conversation on Slack?
Looking forward to hearing your opinion....

**Reply by helicop0420** [reply]
Also One question more.
Can I get source code(svelte) for built-in budibase component?
